clang-tools-extra/test/clang-tidy/checkers/bugprone/unsafe-crtp.cpp

```diff
@@ -0,0 +1,232 @@
+// RUN: %check_clang_tidy %s bugprone-unsafe-crtp %t
+
+namespace class_implicit_ctor {
+template <typename T>
+class CRTP {};
+// CHECK-MESSAGES: :[[@LINE-1]]:7: warning: the implicit default constructor of the CRTP is publicly accessible [bugprone-unsafe-crtp]
+// CHECK-MESSAGES: :[[@LINE-2]]:7: note: consider making it private
+// CHECK-FIXES: CRTP() = default;
+
+class A : CRTP<A> {};
+} // namespace class_implicit_ctor
+
+namespace class_uncostructible {
+template <typename T>
+class CRTP {
+// CHECK-MESSAGES: :[[@LINE-1]]:7: warning: the CRTP cannot be constructed from the derived class [bugprone-unsafe-crtp]
```

Comment:
(Note: These warnings are matched as regexes and substrings, so you can simplify the test file by not repeating the check's name on every warning line.)

Reply:
Some checker test files leave the name of the checker there, some don't, some mixes the two. I think it doesn't hurt to know that the message should be expected from this checker, though it might be obvious based on the name of the test.